test/unit/settings_test.py

```diff
@@ -132,21 +132,18 @@ def test_860m(self):
 class TestEiaSettings:
     """Test pydantic model that validates EIA datasets."""
 
-    # def test_eia923_dependency(self):
-    #     """Test 860 is added if 923 is specified and 860 is not."""
-    #     eia923_settings = Eia923Settings()
-    #     settings = EiaSettings(eia923=eia923_settings)
-    #     data_source = DataSource.from_id("eia860")
-
-    #     assert settings.eia860
-
-    #     assert settings.eia860.years == data_source.working_partitions["years"]
-
-    # def test_eia860_dependency(self):
-    #     """Test 923 tables are added to eia860 if 923 is not specified."""
-    #     eia860_settings = Eia860Settings()
-    #     settings = EiaSettings(eia860=eia860_settings)
-    #     assert settings.eia923.years == eia860_settings.years
+    def test_eia923_dependency(self):
+        """Test that there is some overlap between EIA860 and EIA923 data."""
+        settings = EiaSettings()
+        data_source = DataSource.from_id("eia860")
+        assert settings.eia860
+        # assign both EIA form years
+        eia860_years = settings.eia860.years
+        eia923_years_partition = data_source.working_partitions["years"]
+        eia923_years_settings = settings.eia923.years
+        # assert that there is some overlap between EIA years
+        assert not set(eia860_years).isdisjoint(eia923_years_partition)
+        assert not set(eia860_years).isdisjoint(eia923_years_settings)
```

Comment:
The previous version of these two test were checking if the **default** years of 923 and 860 were getting added in when there was only 860 or 923 data in the settings respectively.

The first `test_eia923_dependency` test was checking this scenario:
* **Previously**: If you fed PUDL settings EIA 923 only, our `EiaSettings` knows that 860 also needs to exist and gives it some default years. Are the 860 years the same as the 923 years?
* **What we could update it to now**: If you fed PUDL settings EIA 923 only, our `EiaSettings` knows that 860 also needs to exist and gives it some default 860 years. Are the 860 years appropriate i.e.: do they have some overlap w/ 923? OR are the assigned default years the years in the partitions?

(honestly i'm not sure now why the previous setup for `test_eia923_dependency` didn't work bc you didn't touch the working partitions for 860 in this PR)

for `test_eia860_dependency`:
* **Previously**: If you fed PUDL settings w/ EIA 860 only, our `EiaSettings` knows that 923 also needs to exist and gives it some default years. Are the 923 years the same as the 860 years?
* **Now**: If you fed PUDL settings w/ EIA 860 only, our `EiaSettings` knows that 923 also needs to exist and gives it some default years. Do the 923 default years at least overlap with the 860 years?


tl;dr version is i think the combining of these two tests removes some important information. Your test builds a default `EiaSettings` object without feeding it anything specific with ends ups triggering `EiaSettings.default_load_all`. The intention previously as I understand it was to test the `EiaSettings.check_eia_dependencies`.

Reply:
Hm ok, my version doesn't combine the two tests (`test_eia860_dependency` and `test_eia923_dependency`) it just combines the two tests that you suggested above? I fixed the weird empty EiaSettings thing which was a mistake on my end. I also re-added the `test_eia860_dependency` test with the same but opposite code from the `test_eia923_dependency`. 

I also had to update the `check_eia_dependencies` in `settings.py` to create default values for eia860 that were based off of whatever was in EIA923 but also what is actually available in 860. It was failing because it was trying to create a default suite of 860 data based on 923 data that included 2023 (which 860 doesn't have). Lmk if this feels like an ok fix.